app/models/cinstance.rb

```diff
@@ -515,5 +515,3 @@ def generate_key
     plan.issuer.prefix_key(SecureRandom.hex(16))
   end
 end
-
-ApplicationContract = Cinstance
```

Comment:
When I removed this alias, I searched for `ApplicationContract` in the code and only found one reference. See the [commit](https://github.com/3scale/porta/pull/3339/commits/dc0c6c39d283499f56d2f65b881db122faa52741). There could always be something like that cucumber step you pointed out, which generates the class name from an interpolated string:

https://github.com/3scale/porta/blob/8bcc7f77bf776619d2eea4a4bc1b1e6b0abfaec9/features/step_definitions/contracts/contract_steps.rb#L9

I'm not sure that test is right, though, because the `type` parameter is transformed by this:

https://github.com/3scale/porta/blob/5c6bbdd999accca358bf16566abe56fdd0ee0c92/features/support/parameter_types.rb#L96-L100

And the resulting `type` can only be `"account_plan"`, `"service_plan"` or `"application_plan"`, which generate class names like `AccountPlanContract`, `ServicePlanContract` and `ApplicationPlanContract`, which don't exist. I think the cucumbers [using](https://github.com/3scale/porta/blob/f6e98d1d8a66f28d13a91ebba8767e12814509ae/features/old/signup/buyers.feature#L39) this step should fail, but they don't, so I assume we can't trust the tests on this. I created a new `ApplicationContract` class in a way it's properly loaded by the autoloader.

Reply:
Hm, but I think in [this step](https://github.com/3scale/porta/blob/8bcc7f77bf776619d2eea4a4bc1b1e6b0abfaec9/features/step_definitions/contracts/contract_steps.rb#L3) the parameter type `{plan_type}` is not used. This step uses a simple regex, so `type` would be either of these values: `account|service|application`. This step is used in this test, I believe: https://github.com/3scale/porta/blob/8bcc7f77bf776619d2eea4a4bc1b1e6b0abfaec9/features/old/signup/buyers.feature#L37-L39

On the other hand, the `plan_type` parameter is used in this step:
https://github.com/3scale/porta/blob/8bcc7f77bf776619d2eea4a4bc1b1e6b0abfaec9/features/step_definitions/plans_steps.rb#L5-L9

I think this is the example step: https://github.com/3scale/porta/blob/8bcc7f77bf776619d2eea4a4bc1b1e6b0abfaec9/features/old/applications/providers/search.feature#L12-L15

So, are you saying that removing `ApplicationContract` alias [this test](https://github.com/3scale/porta/blob/f6e98d1d8a66f28d13a91ebba8767e12814509ae/features/old/signup/buyers.feature#L39) was not failing?...

Reply:
We probably found a bug. I checked it with the debugger, the step receives `type='application_plan'`, when it should receive `type='application'`. Following the stack back to the place where the value is actually set, [here](https://github.com/cucumber/cucumber-expressions/blob/main/ruby/lib/cucumber/cucumber_expressions/argument.rb#L33), `@parameter_type.name` is `'plan_type'`. So the mentioned transformer is being invoked.

> So, are you saying that removing ApplicationContract alias [this test](https://github.com/3scale/porta/blob/f6e98d1d8a66f28d13a91ebba8767e12814509ae/features/old/signup/buyers.feature#L39) was not failing?...

Exactly.